src/marketplace/ListingsView.js

```diff
@@ -52,33 +52,116 @@ class ListingsView extends React.Component<Props, State> {
 		this.setState({userId:jsonObjectOfUser._id});
 	};
 
+
+	//This function directs the view to the proper function
+	//If we are in the tank view it directs it to getMarketSalesForTanks()
+	//Else it directs it to getMarketSalesForComponents()
+	directSaleToProperFunction() : void {
+		if(this.props.sellerType === 'tanks') {
```

Comment:
Wait, what? `sellerType` is of type ComponentType, which is: `'chassis' | 'weapon' | 'scanner' | 'scannerAddon' | 'jammer' | 'treads' | 'item'`. So if this is true, we have a very big problem that is causing your props to not be the type the code thinks it is.